include/swift/AST/DiagnosticEngine.h

```diff
@@ -685,9 +688,8 @@ namespace swift {
     /// but rather stored until all transactions complete.
     llvm::StringSet<llvm::BumpPtrAllocator &> TransactionStrings;
 
-    /// Diagnostic producer to handle the logic behind retriving a diagnostic message,
-    /// from YAML files.
-    Optional<YAMLLocalizationProducer> localization;
+    /// Diagnostic producer to handle the logic behind retriving a localized diagnostic message.
+    Optional<std::unique_ptr<LocalizationProducer>> localization;
```

Comment:
@xedin I made `localization` an `LLVM::Optional` of type `std::unique_ptr`, because if `localization` isn't a pointer, we will experience something called "code slicing". [Check out this answer on StackOverflow.](https://stackoverflow.com/a/4869284/5025731)

Reply:
If you want to allocate `LocalizationProducer` on heap, the you don't really need to use optional, it can simply be `std::unique_ptr<LocalizationProducer>`.

Reply:
So do you suggest to use to remove `Optional`?

Reply:
Yes, that's is correct.